tests/nightly/estimator/test_sentiment_rnn.py

```diff
@@ -0,0 +1,272 @@
+
+    return [tokenizer(review) for review, _ in data]
+
+
+def get_vocab_imdb(data):
+    '''
+    Get the indexed tokens
+    '''
+    tokenized_data = get_tokenized_imdb(data)
+    counter = collections.Counter([tk for st in tokenized_data for tk in st])
+    return text.vocab.Vocabulary(counter, min_freq=5)
+
+
+def preprocess_imdb(data, vocab):
+    '''
+    Make the length of each comment 500 by truncating or adding 0s
+    '''
+    max_l = 500
+
+    def pad(x):
+        return x[:max_l] if len(x) > max_l else x + [0] * (max_l - len(x))
+
+    tokenized_data = get_tokenized_imdb(data)
+    features = nd.array([pad(vocab.to_indices(x)) for x in tokenized_data])
+    labels = nd.array([score for _, score in data])
+    return features, labels
+
+
+def test_estimator_cpu():
+    '''
+    Test estimator by doing one pass over each model with synthetic data
+    '''
+    models = ['TextCNN', 'BiRNN']
+    context = mx.cpu()
+    batch_size = 64
+    num_epochs = 1
+    lr = 0.01
+    embed_size = 100
+
+    train_data = mx.nd.random.randint(low=0, high=100, shape=(2 * batch_size, 500))
+    train_label = mx.nd.random.randint(low=0, high=2, shape=(2 * batch_size,))
+    val_data = mx.nd.random.randint(low=0, high=100, shape=(batch_size, 500))
+    val_label = mx.nd.random.randint(low=0, high=2, shape=(batch_size,))
+
+    train_dataloader = gdata.DataLoader(dataset=gdata.ArrayDataset(train_data, train_label),
+                                        batch_size=batch_size, shuffle=True)
+    val_dataloader = gdata.DataLoader(dataset=gdata.ArrayDataset(val_data, val_label),
+                                      batch_size=batch_size)
+    vocab_list = mx.nd.zeros(shape=(100,))
+
+    # Get the model
+    for model in models:
+        if model == 'TextCNN':
+            kernel_sizes, nums_channels = [3, 4, 5], [100, 100, 100]
+            net = TextCNN(vocab_list, embed_size, kernel_sizes, nums_channels)
+        else:
+            num_hiddens, num_layers = 100, 2
+            net = BiRNN(vocab_list, embed_size, num_hiddens, num_layers)
+        net.initialize(mx.init.Xavier(), ctx=context)
+        # Define trainer
+        trainer = mx.gluon.Trainer(net.collect_params(), 'adam', {'learning_rate': lr})
+        # Define loss and evaluation metrics
+        loss = gloss.SoftmaxCrossEntropyLoss()
+        acc = mx.metric.Accuracy()
+
+        # Define estimator
+        e = est.Estimator(net=net, loss=loss, metrics=acc,
+                          trainers=trainer, context=context)
+        # Begin training
+        e.fit(train_data=train_dataloader, val_data=val_dataloader,
+              epochs=num_epochs, batch_size=batch_size)
+
+
+def test_estimator_gpu():
+    '''
+    Test estimator by training Bidirectional RNN for 5 epochs on the IMDB dataset
+    and verify accuracy
+    '''
+    batch_size = 64
+    num_epochs = 5
+    lr = 0.01
+    embed_size = 100
+
+    # Set context
+    if mx.context.num_gpus() > 0:
```

Comment:
Can write this more succinctly as ```ctx = mx.gpu() if mx.context.num_gpus() > 0 else mx.cpu()```

Reply:
Yes absolutely. Modified it. [ae81b08](https://github.com/apache/incubator-mxnet/pull/14547/commits/ae81b08fd94686d0501fe4e3c3762f3fef154db5)